File: src/LMS_Assignment.Domain/Entities/Assignment.cs
using LMS_Assignment.Domain.Enums;

namespace LMS_Assignment.Domain.Entities;

public class Assignment
{
    public Guid Id { get; set; }
    public Guid TeacherSubjectAssignmentId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DateTime Deadline { get; set; }
    public decimal MaxMarks { get; set; }
    public bool AllowResubmission { get; set; } = true;
    public AssignmentStatus Status { get; set; } = AssignmentStatus.Draft;
    public bool IsDeleted { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public TeacherSubjectAssignment TeacherSubjectAssignment { get; set; } = null!;
    public ICollection<AssignmentAttachment> Attachments { get; set; } = new List<AssignmentAttachment>();
    public ICollection<Submission> Submissions { get; set; } = new List<Submission>();
}

File: frontend/src/types/assignment.ts
export type AssignmentStatus = "Draft" | "Published"

export interface Assignment {
  id: string
  teacherSubjectAssignmentId: string
  teacherName: string
  className: string
  subjectName: string
  title: string
  description: string
  deadline: string
  maxMarks: number
  allowResubmission: boolean
  status: AssignmentStatus
}

export interface CreateAssignmentRequest {
  teacherSubjectAssignmentId: string
  title: string
  description: string
  deadline: string
  maxMarks: number
  allowResubmission: boolean
}

export interface UpdateAssignmentRequest {
  title: string
  description: string
  deadline: string
  maxMarks: number
  allowResubmission: boolean
}
